src/Illuminate/View/Engines/CompilerEngine.php

```diff
@@ -51,10 +58,12 @@ public function get($path, array $data = [])
         // If this given view has expired, which means it has simply been edited since
         // it was last compiled, we will re-compile the views so we can evaluate a
         // fresh copy of the view. We'll pass the compiler the path of the view.
-        if ($this->compiler->isExpired($path)) {
+        if (! isset($this->compiledOrNotExpired[$path]) && $this->compiler->isExpired($path)) {
```

Comment:
@nunomaduro this is giving me a Exception https://github.com/spatie/laravel-permission/actions/runs/3276093280/jobs/5391731843 after `9.36.0`
maybe must be
```diff
-  if (! isset($this->compiledOrNotExpired[$path]) && $this->compiler->isExpired($path)) {
+  if (! isset($this->compiledOrNotExpired[$path]) || $this->compiler->isExpired($path)) {
```
Or i'm doing something wrong, nothing changed on that package

btw, thanks for the great work

Reply:
Fixed: https://github.com/laravel/framework/pull/44643.